File: Chapter 2/InsertionSort.java
public class InsertionSort {

    public static void main(String[] args){
        int[] a = {5,2,4,6,1,3};
        int[] b = {31, 41, 59, 26, 41, 58};

        arrayPrinter(a);
        insertionSortAscending(a);
        insertionSortDescending(a);
        arrayPrinter(b);
        insertionSortAscending(b);
        insertionSortDescending(b);
    }

    public static void arrayPrinter(int[] array){
        for(int i = 0; i < array.length; i++){
            System.out.print(array[i] + "\t");
        }
        System.out.println("\n");
    }

    public static void insertionSortAscending(int[] a){
        int key, i;
        for(int j = 1; j < a.length; j++){
            key = a[j];
            i = j - 1;
            while(i >= 0  && a[i] > key){
                a[i + 1] = a[i];
                i--;
            }
            a[i + 1] = key;
        }
        arrayPrinter(a);
    }

    public static void insertionSortDescending(int[] a){
        int key, i;
        for(int j = 1; j < a.length; j++){
            key = a[j];
            i = j - 1;
            while(i >= 0 && a[i] < key){
                a[i+1] = a[i];
                i--;
            }
            a[i+1] = key;
        }
        
        arrayPrinter(a);
    }

}

File: Chapter 2/InsertionSort.js
let a = [5, 2, 4, 6, 1, 3]
let b = [31, 41, 59, 26, 41, 58]

function insertionSortAscending(a){
    let key, i
    for(let j = 1; j < a.length; j++){
        key = a[j]
        i = j - 1
        while(i >= 0 && a[i] > key){
            a[i + 1] = a[i]
            i--
        }
        a[i+1] = key
    }
    console.log(a)
}

function insertionSortDescending(a){
    let key, i
    for(let j = 1; j < a.length; j++){
        key = a[j]
        i = j - 1
        while(i >= 0 && a[i] < key){
            a[i+1] = a[i]
            i--
        }
        a[i+1] = key
    }
    console.log(a);
}

console.log(a)
insertionSortAscending(a)
insertionSortDescending(a)
console.log(b)
insertionSortAscending(b)
insertionSortDescending(b)